### ISSUE TITLE
Volume range seems logarithmic, and a precise-volume bug

### ISSUE THREAD

**Issue by francislavoie** [th-ch/youtube-music#500]
Kinda strange, changing the volume anywhere between ~35-100% seems to do nothing, but the volume levels between 0-35% are basically the full range. So changing the volume from 100% to 50% does basically nothing.

This behaviour is very different from https://github.com/ytmdesktop/ytmdesktop which more gradually changes volume.

---

Didn't feel like making a separate issue for this, but the `precise-volume` plugin (which I enabled to see if it would make any difference to the volume range mentioned above -- it doesn't) has a problem with the "Set Custom Volume Steps" dialog window; it flickers into view for one frame, then automatically closes before I can interact with it. This is on Windows 10. Other dialog windows like the one from the `shortcuts` plugin opens up fine though. I was able to change the steps via the `config.json` file so it's not too big a deal for me.

Wild guess from looking at the code and comparing `shortcuts` to `precise-volume`, is it because there's no defined `height`?

**Comment by Araxeus** [comment]
> Didn't feel like making a separate issue for this, but the `precise-volume` plugin (which I enabled to see if it would make any difference to the volume range mentioned above -- it doesn't) has a problem with the "Set Custom Volume Steps" dialog window; it flickers into view for one frame, then automatically closes before I can interact with it. This is on Windows 10. Other dialog windows like the one from the `shortcuts` plugin opens up fine though. I was able to change the steps via the `config.json` file so it's not too big a deal for me.
> 
> Wild guess from looking at the code and comparing `shortcuts` to `precise-volume`, is it because there's no defined `height`?

Thank you for this report! this actually happened because of a bug in the `custom-electron-prompt` library which I fixed and updated that package to the new version in PR #476 commit abc1712

because of a mistake caused by an external pr, some function was missing an argument - which was fixed in the library easily
https://github.com/Araxeus/custom-electron-prompt/commit/dd5fc8e982bec061a3854e3440e4e6fa59e72756

**Comment by Araxeus** [comment]
> Kinda strange, changing the volume anywhere between ~35-100% seems to do nothing, but the volume levels between 0-35% are basically the full range. So changing the volume from 100% to 50% does basically nothing.

I can't actually reproduce this, it all works fine for me. but maybe you should try the new upcoming `Exponential Volume` plugin when it comes out after #488 is merged

I think that plugin might help with your problem 🙂

**Comment by francislavoie** [comment]
Ah nice! Yeah, that new plugin looks like exactly what I was hoping for. Thanks :+1: I'll close this since it's already being tracked in that PR.
